host/sgx/exception_sim.c

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/host.h>
+#include <openenclave/internal/registers.h>
+#include <openenclave/internal/sgx/td.h>
+#include <signal.h>
+#include <ucontext.h>
+#include "enclave.h"
+#include "exception.h"
+
+bool is_simulation(void)
+{
+    // In simulation mode the fs still points to td.
+    oe_sgx_td_t* td = (oe_sgx_td_t*)oe_get_fs_register_base();
```

Comment:
Can you add the following scenario to the test:
- Add a host side exception handler
- Introduce a host side exception that will be handled by the above handler.

I'm curious whether the expression `td->magic == TD_MAGIC` is safe when FS contains host-side FS value (ie when the exception happens in the host)

Reply:
It's probably safer to keep a host-side global for simulation mode right? This also avoids differences in how fs is managed on Linux and Windows.

Reply:
> I'm curious whether the expression td->magic == TD_MAGIC is safe when FS contains host-side FS value (ie when the exception happens in the host)

I agree that if we are worried about fs possibly being changed then this de-reference is unsafe.

Reply:
> It's probably safer to keep a host-side global for simulation mode right? This also avoids differences in how fs is managed on Linux and Windows.

Yes, a global (spin-lock guarded or lock-free) linked-list that gets populated only for simulation mode might be the most reliable option.

Reply:
Dumb question: shouldn't only a single thread be accessing `td` at any given point? A second thread with a different FS encountering a host exception wouldn't reference the same FS base or TD, no?